src/charm.py

```diff
@@ -23,82 +24,40 @@ class FlogCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
 
+        # Relation managers
+        # NOTE: these are expected to be fully initialized here, i.e. in this approach, relation managers do not observe
+        # any events - they are fully initialized after their constructor is done.
         self._log_proxy = LogProxyConsumer(
             charm=self, log_files=["/bin/fake.log"], container_name="workload"
         )
+        # TODO refactor LogProxyConsumer so it has a `.manifest()` getter we could apply it using the workload instance.
+
+        manifest = flog_manifest(self.config)
```

Comment:
is this when we go from "common exit hook" to "common entry hook"?

Reply:
Haha, yes!
1. I would like to see how this pattern looks, without the distraction of observers. After the basics are nailed down I will look into gating with observers.
2. Without [key juju events that are currently missing](https://discourse.charmhub.io/t/new-juju-events-i-would-love-to-have/10952) we'd still need the common exit hook anyway.